### PR TITLE
Compatibility upgraded to HWI 2.0.2 binary along with fixes

### PR DESCRIPTION
**Pull request by wszdexdrf**
Updated flags and commands to match the current version of the HWI binary. For example `--testnet` flag to `--chain` arguments etc.

### CODE REVIEW

**Comment by wszdexdrf** [comment]
I found out the problem with the Trezor `test_sign_message`. I think it was due to the Derivation Paths not conforming to the latest standards. It should work now.

**Comment by danielabrozzoni** [comment]
I'm closing this one, as it's better if we keep working on #9

**Comment by danielabrozzoni** [comment]
Nevermind, for some reason I can't reopen #9, let's remain here...

**Review by danielabrozzoni — COMMENTED** [review]
Looking good, we just need to improve history a bit.
For example:
1. 37d04374189b63b1d7ce6f1e90dab493eaf614f2 is called "HWIPartiallySignedTransaction contains a PartiallySignedTransaction", but it's doing more: it's also adding `test_sign_tx`. In this case `test_sign_tx` is unrelated, so it should have been added in a different commit. If the change is related, you should put in the commit description that you're also doing that specific change.
2. 4ef45896873a788f798d2193766abb95488c9fb7 removes test_sign_tx. It shouldn't be a commit by itself, since `test_sign_tx` has been added in this PR: you can just remove it from the commit that added it.
3. 123ec3cf951ffa0e2a6200cdaeda4af887c170ac is called "Fixed test sign_message", but it's doing much more: it's also commenting `test_sign_tx`, renaming `RegTest` and `SigNet` in `regtest` and `signet`, etc

But don't worry! We can fix these problems easily :) 

First, we'll reset everything to `8975a5ecce1c21aa8563ec0f86ca668b5d7fafe4`. Since it's a soft reset, it won't delete your changes:
```
git reset --soft 8975a5ecce1c21aa8563ec0f86ca668b5d7fafe4
```

Now, if you `git status`:
```
❯ git status
Changes to be committed:
  (use "git restore --staged <file>..." to unstage)
        modified:   src/commands.rs
        modified:   src/interface.rs
        modified:   src/lib.rs
        modified:   src/types.rs
```
All this changes are ready to be committed, but we actually want to create different commits. We unstage them:
```
git restore --staged .
```
And now, `git status` shows:
```
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   src/commands.rs
        modified:   src/interface.rs
        modified:   src/lib.rs
        modified:   src/types.rs
```
Run `cargo fmt`, so all the code will be formatted properly.

Now, we want to create different commits:
- "HWISignature contains a Vec<u8> instead of String"
- "HWIPartiallySignedTransaction contains a PartiallySignedTransaction"
- "Implement ExtendedPubKey for HWIExtendedPubKey" (note: you originally called it "implemented ...", but it's better to use **implement**, as you should always use the imperative in commits, see https://www.theserverside.com/video/Follow-these-git-commit-message-guidelines)
- "Compatibility upgrade to HWI 2.0.2": this commit will contain all the other changes, which will be listed in the body of the commit (fix test_sign_message to work with trezor, small fixes in the tests) 

How do you create the different commits? You use `git add -p *file*` to stage parts of the file, then you `git commit`.

Most of the time you open a PR people will request small changes to your code. Instead of creating a new commit for each small change, you should learn how to use interactive rebases. 
What I usually do is: let's say I have to change something in commit `1234`. I make the changes, I add them, then I run:
```
git commit --fixup 1234
```
this will create a commit called `!fixup: *title of commit 1234`. 
I then run
```
git rebase -i origin/master --autosquash
```
This will open the interactive rebase tool (don't worry too much about it for now, you'll learn how to use it), which will show you that it's taking your last commit and squashing it into `1234`. Perfect. Now just exit the editor, and everything should be fine (check in the `git log`: you shouldn't have the `!fixup` commit, you should have the change in `1234`).
If you want to learn more:
https://thoughtbot.com/blog/git-interactive-rebase-squash-amend-rewriting-history

File: src/types.rs [hunk 1]
```diff
@@ -20,7 +45,22 @@ pub struct HWIAddress {
 
 #[derive(Deserialize)]
 pub struct HWIPartiallySignedTransaction {
-    pub psbt: String,
+    #[serde(deserialize_with="deserialize_psbt")]
```
[the remark this answers is not in the document]
Comment (wszdexdrf) [reply src/types.rs:48]: Yes, you are right. I hadn't added the base64 feature flag in cargo.toml, therefore it wasn't working for me.
As a side note, I think there isn't any test checking `HWIPartiallySignedTransaction`'s functionality. The only one was `sign_tx()`, which I removed. XD

**Comment by wszdexdrf** [comment]
Fixed everything.

**Review by danielabrozzoni — APPROVED** [review]
ACK 7def42fe4559becca72a013d4c678dad4ee08a85

Sorry this took so long!

**Pull request merged**